### PR TITLE
update requested capacity card for external mode

### PR DESCRIPTION
**Pull request by SanjalKatiyar**
refactor of https://github.com/red-hat-storage/odf-console/pull/1000 (closed original PR)
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2228375

Screenshot 2023-09-05 at 12 57 56 PM

### CODE REVIEW

File: packages/ocs/dashboards/persistent-internal/capacity-breakdown-card/capacity-breakdown-card.tsx [hunk 1]
```diff
@@ -54,6 +55,74 @@ const modelByTotalQueryMap = {
     StorageDashboardQuery.PVC_NAMESPACES_TOTAL_USED,
 };
 
+const getInitialSortedNs = (
+  allResources: K8sResourceCommon[]
+): K8sResourceCommon =>
+  allResources?.sort((a, b) =>
+    a.metadata.name.localeCompare(b.metadata.name)
```
Comment (GowthamShanmugam) [inline packages/ocs/dashboards/persistent-internal/capacity-breakdown-card/capacity-breakdown-card.tsx:62]: is it ok to use getName?

File: packages/ocs/dashboards/persistent-internal/capacity-breakdown-card/capacity-breakdown-card.tsx [hunk 2]
```diff
@@ -54,6 +55,74 @@ const modelByTotalQueryMap = {
...
+  )[0];
```
[the remark this answers is not in the document]
Comment (SanjalKatiyar) [reply packages/ocs/dashboards/persistent-internal/capacity-breakdown-card/capacity-breakdown-card.tsx:63]: if "allResources" is empty, it should not run the "sort" function at all... so we should be good IMO...

File: packages/ocs/dashboards/persistent-internal/capacity-breakdown-card/capacity-breakdown-card.tsx (hunk 1 above)
Comment (SanjalKatiyar) [reply packages/ocs/dashboards/persistent-internal/capacity-breakdown-card/capacity-breakdown-card.tsx]: updated...

File: packages/ocs/dashboards/persistent-internal/capacity-breakdown-card/capacity-breakdown-card.tsx [hunk 3]
```diff
@@ -54,6 +55,72 @@ const modelByTotalQueryMap = {
     StorageDashboardQuery.PVC_NAMESPACES_TOTAL_USED,
 };
 
+const getInitialSortedNs = (
+  allResources: K8sResourceCommon[]
+): K8sResourceCommon =>
+  allResources?.sort((a, b) => getName(a).localeCompare(getName(b)))[0];
+
+export const NamespaceDropdown: React.FC<{
```
Comment (bipuladh) [inline packages/ocs/dashboards/persistent-internal/capacity-breakdown-card/capacity-breakdown-card.tsx:63]: wasn't this a generic component before hand?

Comment (SanjalKatiyar) [reply packages/ocs/dashboards/persistent-internal/capacity-breakdown-card/capacity-breakdown-card.tsx:63]: I think we had it in console, but later on our side we implemented `ResourceDropdown` instead for our use case...
this FC is using `ResourceDropdown` as well, just few classes on top of it...

Comment (SanjalKatiyar) [reply packages/ocs/dashboards/persistent-internal/capacity-breakdown-card/capacity-breakdown-card.tsx:63]: I just moved the code out to a separate FC in this PR to reuse in external card, earlier this code was inline with `BreakdownCard` FC...

**Review by GowthamShanmugam — APPROVED** [review]

**Review by GowthamShanmugam — APPROVED** [review]

**Pull request merged**